### PR TITLE
WIP: Adding support for unknown labels

### PR DESCRIPTION
**Pull request by rdedhia**

### CODE REVIEW

File: finetune/target_models/sequence_labeling.py [hunk 1]
```diff
@@ -217,25 +275,75 @@ def finetune(self, Xs, Y=None, context=None, update_hook=None):
...
+                self.config.max_empty_chunk_ratio = 0.0
+
+        super().finetune(Xs, Y=Y, context=context, update_hook=update_hook)
+
+        # This is hacky and there's probably a better way to do this
+        # This is required because otherwise the tensor shapes do not match during prediction
+        unknown_token = self.config["unknown_token"]
+        if (
+            self.config["unknown_labels"]
+            and unknown_token in self.input_pipeline.label_encoder.target_labels
```
Comment (madisonmay) [inline finetune/target_models/sequence_labeling.py:356]: Maybe some of these -1's could be handled by making more of this a responsibility of the label encoder?

**Comment by madisonmay** [comment]
Looks correct to me.  Some places where tf code could be made slightly cleaner but not even worth sweating that yet since we don't even know whether we'll be keeping those code changes until we get more experimental results.

**Comment by madisonmay** [comment]
We probably won't merge for a while but nice to have a place to get WIP diffs to look at.  Thanks man!

File: finetune/target_models/sequence_labeling.py (hunk 1 above)
Comment (rdedhia) [reply finetune/target_models/sequence_labeling.py:356]: Yeah I like that idea. Basically I'm running into a scenario where I need the extra `<UNK>` dim whenever I train, but not when I predict. And the code gets hairier anytime there's multiple train/predict loops (like w/ ANS or partial labels creation)

**Comment by madisonmay** [comment]
Note: remove unused cruft and merge smaller changeset in.

**Comment by rdedhia** [comment]
Closing because unknown labels experiment did not work. Will make a separate PR for partial labels ANS changes

**Pull request closed without merging**